src/data/data_fetcher.py

```diff
@@ -249,31 +236,32 @@ def _get_transactions_on_page(self):
         Gets last_index and last_contribution_receipt_date to pass to get_next_page.
         Returns complete_list with transactions from page.
         """
+
+        
         # Pull out the data we want from each transaction on a page and add it to the complete_list
         for item in self.info["results"]:
-            party = item["committee"]["party"]
-            contributor_occupation = item["contributor_occupation"]
-            contributor_employer = item["contributor_employer"]
-            contributor_city = item["contributor_city"]
-            contributor_state = item["contributor_state"]
-            contributor_zip = item["contributor_zip"]
-            # Because there is stupid data in some transactions for ZIP, need to do this for all data to process for bad data
             try:
-                if contributor_zip.isnumeric():
+                if contributor_zip := item["contributor_zip"].isnumeric():
+                    if len(contributor_zip) < 5:
+                        contributor_zip = 99999
+                    elif len(contributor_zip) > 5:
+                        contributor_zip = int(contributor_zip[0:5])
                     contributor_zip = int(contributor_zip)
                 else:
-                    contributor_zip = int("00000")
+                    contributor_zip = 99999
             except:
-                contributor_zip = int("00000")
-            current_list = [
-                contributor_occupation,
-                contributor_employer,
-                contributor_city,
-                contributor_state,
+                contributor_zip = 99999
+            
+            self.complete_list.append(
+            current_list := [
+                contributor_occupation := item["contributor_occupation"],
+                contributor_employer := item["contributor_employer"],
+                contributor_city := item["contributor_city"],
+                contributor_state := item["contributor_state"],
                 contributor_zip,
-                party,
+                party := item["committee"]["party"],
```

Comment:
S P I C Y

I'm still afraid of this thing, lol

Reply:
I have been optimizing for speed because I got the FEC to upgrade my API requests to 120 per min. Walrus and string formatting was a ridiculous improvement. Went from about two seconds per page to pulling two pages in one second.

Reply:
Might not be applicable here, but eyeballing this section, it *might* be a good use of your time to chew on this post awhile.

https://engineering.upside.com/a-beginners-guide-to-optimizing-pandas-code-for-speed-c09ef2c6a4d6

I'd **absolutely** suggest writing unit tests first, before you go a'changing things, but learning how to performance tune my `numpy/pandas` data code was a seriously-rewarding rabbit hole I went down a couple years ago and am damn glad I did. This post is an all-timer for me, tbh